test/boost/stall_free_test.cc

```diff
@@ -214,22 +216,29 @@ SEASTAR_THREAD_TEST_CASE(test_clear_gently_foreign_shared_ptr) {
             return make_foreign<lw_shared_ptr<clear_gently_tracker<int>>>(std::move(p));
         }).get();
     };
-    foreign_ptr<lw_shared_ptr<clear_gently_tracker<int>>> p0 = make_foreign_ptr();
+    std::array<foreign_ptr<lw_shared_ptr<clear_gently_tracker<int>>>, 2> ptrs;
+    auto& p0 = ptrs[0];
+    p0 = make_foreign_ptr();
 
     utils::clear_gently(p0).get();
-    BOOST_CHECK(p0);
+    BOOST_REQUIRE(!p0);
     BOOST_REQUIRE_EQUAL(cleared_gently, 1);
 
-    p0 = make_foreign_ptr();
-    auto p1 = p0.copy().get();
-
+    // Test re-clearing the already-reset foreign shared ptr
     utils::clear_gently(p0).get();
-    BOOST_REQUIRE_EQUAL(cleared_gently, 1);
-    utils::clear_gently(p1).get();
+    BOOST_REQUIRE(!p0);
     BOOST_REQUIRE_EQUAL(cleared_gently, 1);
 
-    p0.reset();
-    utils::clear_gently(p1).get();
+    // Test clearing of foreign shared ptrs
```

Comment:
This comment is a bit out of place because the lines above also checked clearing of foreign shared pointers. Maybe you mean that you are testing here the case of a shared pointers having **more than one references**?

Reply:
Right  I'll try to clarify